core/multiproc/inc/TProcPool.h

```diff
@@ -45,28 +43,12 @@ class TProcPool : private TMPClient {
    TProcPool &operator=(const TProcPool &) = delete;
 
    // Map
-   //these late return types allow for a compile-time check of compatibility between function signatures and args,
-   //and a compile-time check that the argument list implements a front() method (all STL sequence containers have it)
    template<class F> auto Map(F func, unsigned nTimes) -> std::vector<decltype(func())>;
-   template<class F, class T> auto Map(F func, T &args) -> std::vector < decltype(++(args.begin()), args.end(), func(args.front())) >;
-   /// \cond doxygen should ignore these methods
-   template<class F> TObjArray Map(F func, TCollection &args);
-   template<class F, class T> auto Map(F func, std::initializer_list<T> args) -> std::vector<decltype(func(*args.begin()))>;
+   /// \cond
+   template<class F, class INTEGER> auto Map(F func, ROOT::TSeq<INTEGER> args) -> std::vector<decltype(func(args.front()))>;
```

Comment:
Will check if the function returns a reference with an extra argument in the template, as seen [here](https://root.cern.ch/gitweb/?p=root.git;a=blob;f=hist/hist/v7/inc/ROOT/THist.h;h=8f119f32e80b854ea335750820dc8514e8ad9d89;hb=HEAD#l318). For example:  `template<class F, class T,
    class noReferenceCond = typename std::enable_if<"Function can't return a reference" && !(std::is_reference<std::result_of<F(T)>>::value)>>`